tools/sftd_disco/sftd_disco.sh

```diff
@@ -30,23 +28,23 @@ function valid_url() {
 # this file can then be served from nginx running besides sft
 function upstream() {
     name=$1
-    port=${2:-'8585'}
-    entries=$(dig +short +retries=3 +search SRV ${name} | sort)
+    entries=$(dig +short +retries=3 +search SRV "${name}" | sort)
     unset servers
     comma=""
     IFS=$'\n'
-    for entry in ${entries[@]}; do
+    for entry in "${entries[@]}"; do
```

Comment:
**issue:** same thing here: the array is expanded inside the quotes, so you just get one `entry` with all records inside.

**suggestion:** don't quote `${entries[@]}`, however use quoting when using `$entry` (as you did down the line).

```suggestion
    for entry in ${entries[@]}; do
```

Reply:
Thank you for checking! No, actually array quoted arrays are still splitting, and shellcheck suggests it https://www.shellcheck.net/wiki/SC2068
The problem here is that entries is not array, but a string. Applied your suggestsions in https://github.com/wireapp/wire-server/pull/4220/commits/88b16c876cfb10ca69969812616476fd33112c13